### PR TITLE
Basic Tari Universe provider

### PR DESCRIPTION
**Pull request by MCozhusheck**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -11,11 +11,11 @@ import {
   TemplateDefinition,
   SubstateRequirement,
   Substate,
-} from './providers/types'
-import {MetamaskTariProvider}  from "./providers/metamask";
-import { MetaMaskInpageProvider } from '@metamask/providers';
-import {WalletDaemonTariProvider, WalletDaemonParameters} from "./providers/wallet_daemon";
-import {TariPermissions} from "./providers/wallet_daemon";
+} from "./providers/types";
```
Comment (kowalski) [inline src/index.ts:14]: try to avoid changing linter conventions in existing codebase
we shouldn't be touching this line

File: src/providers/tari_universe/provider.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  }
+
+  public async getAccount(): Promise<Account> {
+    const { account, public_key } = (await this.sendRequest({ methodName: "getAccount", args: [] })) as any;
+
+    return {
+      account_id: account.key_index,
+      address: account.address.Component,
+      public_key,
+      // TODO
```
Comment (kowalski) [inline src/providers/tari_universe/provider.ts:81]: add link to issue

File: src/providers/types.ts [hunk 3]
```diff
@@ -1,68 +1,74 @@
...
 
 export enum TransactionStatus {
-    New,
-    DryRun,
-    Pending,
-    Accepted,
-    Rejected,
-    InvalidTransaction,
-    OnlyFeeAccepted
+  New,
```
Comment (kowalski) [inline src/providers/types.ts:31]: more unnecessary changes to linter conventions - please revert this to original state

File: src/providers/tari_universe/provider.ts [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  name?: string;
+  onConnection?: () => void;
+};
+
+export class TariUniverseProvider implements TariProvider {
+  public providerName = "WalletDaemon";
+  params: TariUniverseProviderParameters;
+  private __id = 0;
+
+  public constructor(params: TariUniverseProviderParameters) {
```
Comment (kowalski) [inline src/providers/tari_universe/provider.ts:30]:
```
- params: TariUniverseProviderParameters;
+ public constructor(public params: TariUniverseProviderParameters) {}
```

no need for assignment when you put "public" directive in params

File: src/providers/tari_universe/provider.ts [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+  Account,
+  TariUniverseProviderRequest,
+  Substate,
+  TemplateDefinition,
+  VaultBalances,
+} from "../types";
+import { TariProvider } from "../index";
+import { Instruction, TransactionSubmitRequest, AccountsGetBalancesResponse } from "@tariproject/wallet_jrpc_client";
+
+export const Unsupported = "UNSUPPORTED";
```
Comment (karczuRF) [inline src/providers/tari_universe/provider.ts:16]: Where is it used? As I see the same const is defined in `wallet_daemon/provider.ts` but also not used. Remove it or create new file with constants to not duplicate this one

**Review by karczuRF — APPROVED** [review]

File: src/providers/types.ts [hunk 6]
```diff
@@ -1,68 +1,85 @@
...
+export type ProviderRequest<T extends ProviderMethodNames> = {
+  id: number;
+  methodName: T;
+  args: Parameters<ProviderMethods[T]>;
+};
+
+export type ProviderResponse<T extends ProviderMethodNames> = {
+  id: number;
+  result: ProviderReturnType<T>;
+};
```
Comment (mrnaveira) [inline src/providers/types.ts:85]: It seems to me that these new types are only meant for the new Tari Universe provider.

If that's the case, then this type definitions should not be located here in the common provider types, but encapsulated inside the `tari_universe` folder.

Comment (MCozhusheck) [reply src/providers/types.ts]: Done

File: src/providers/wallet_daemon/provider.ts [hunk 7]
```diff
@@ -22,7 +22,6 @@ import {
 import {WebRtcRpcTransport} from "./webrtc_transport";
 
 export const WalletDaemonNotConnected = 'WALLET_DAEMON_NOT_CONNECTED';
-export const Unsupported = 'UNSUPPORTED';
```
Comment (kowalski) [inline src/providers/wallet_daemon/provider.ts:25]: I'm not convinced we should be deleting this line in this PR.
Can we revert changes to files under src/providers/wallet_daemon ?

Comment (MCozhusheck) [reply src/providers/wallet_daemon/provider.ts]: It wasn't used anywhere but since this PR is about only Tari Universe provider I have reverted this change

**Review by mrnaveira — APPROVED** [review]

**Review by sdbondi — APPROVED** [review]

**Comment by sdbondi** [comment]
I think we need a prettierrc in this repo and CI to check them - happy to merge
